File: javascript/algorithms/sort/selection_sort.js
// still O(n^2) time complexity and O(1) space complexity

function swap(array, index1, index2) {
    let temp = array[index1];
    array[index1] = array[index2];
    array[index2] = temp;
}

function selection_sort(items) {
    let len = items.length
    let min;

    for (let i = 0; i < len; i++) {
        min = i;
        for (let j = i+1; j < len; j++) {
            if (items[j] < items[min]) {
                min = j;
            }
        }

        if (i != min) {
            swap(items, i, min);
        }
    }
    console.log(items)
    return items;
}

selection_sort([6,1,23,4,2,3]); // [1, 2, 3, 4, 6, 23]

File: python/algorithms/arrays/sort/selection_sort.py
def swap(array, index1, index2):
    temp = array[index1]
    array[index1] = array[index2]
    array[index2] = temp

def insertion_sort(array):
    arr_length = len(array)

    for i in range(arr_length):
        min_val = i
        for j in range(i+1, arr_length):
            if array[min_val] > array[j]:
                min_val = j
        
        if (min_val != i):
            swap(array, i, min_val)
    print(array)
    return array

test_array = [5,2,6,3,4567,54,23,6,1]
insertion_sort(test_array)